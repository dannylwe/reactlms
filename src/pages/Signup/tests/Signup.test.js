import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import { mount, shallow } from "enzyme";
import SignupDD, { Signup } from "../Signup";
import { Provider } from "react-redux";
import store from "../../../../store";

describe("signup", () => {
  global.alert = jest.fn();
  it("should render", () => {
    const wrapper = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <SignupDD />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it("should handle changes", () => {
    const wrapper = shallow(<Signup />);
    const spy = jest.spyOn(wrapper.instance(), "handleChange");
    wrapper
      .find("#usernameInput")
      .simulate("change", { target: { username: "test_username" } });
    wrapper
      .find("#confirmPasswordlInput")
      .simulate("change", { target: { confirmPassword: "test_password1" } });
    wrapper
      .find("#passwordlInput")
      .simulate("change", { target: { password: "test_password" } });
    wrapper
      .find("#habdphoneInput")
      .simulate("change", { target: { handphone: "0772504991" } });
    wrapper
      .find("#emailInput")
      .simulate("change", { target: { email: "test_email@gmail.com" } });
    expect(spy).toHaveBeenCalled();
  });
  it("should call submit", () => {
    const props = {
      signupAuthentication: jest.fn()
    };
    const wrapper = shallow(<Signup {...props} />);
    const spy = jest.spyOn(wrapper.instance(), "onSubmit");
    const event = { preventDefault: jest.fn() };
    wrapper.instance().onSubmit(event);
    expect(spy).toHaveBeenCalled();
  });
  it("should call check password", () => {
    const props = {
      signupAuthentication: jest.fn()
    };
    const wrapper = shallow(<Signup {...props} />);
    const spy = jest.spyOn(wrapper.instance(), "passwordCheck");
    wrapper
      .find("#usernameInput")
      .simulate("change", { target: { username: "test_username" } });
    wrapper
      .find("#confirmPasswordlInput")
      .simulate("change", { target: { confirmPassword: "test_password1" } });
    const event = { preventDefault: jest.fn() };
    wrapper.instance().onSubmit(event);
    expect(spy).toHaveBeenCalled();
  });
});
