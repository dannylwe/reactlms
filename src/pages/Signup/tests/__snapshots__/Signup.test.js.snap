// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signup should render 1`] = `
Array [
  <div
    className="signup-title"
  >
    <a
      href="/login"
      onClick={[Function]}
    >
      <h1>
        Welcome to SendIT
      </h1>
    </a>
  </div>,
  <div
    className="signup-form"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="signup-forms input-forms"
      >
        <div
          className="email-input"
        >
          <label>
            Email
          </label>
          <input
            aria-describedby="emailHelp"
            className="form-control"
            id="emailInput"
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="email"
          />
        </div>
        <div
          className="username-input"
        >
          <label>
            Username
          </label>
          <input
            aria-describedby="usernameHelp"
            className="form-control"
            id="usernameInput"
            name="username"
            onChange={[Function]}
            placeholder="Username"
            type="text"
          />
        </div>
        <div
          className="phone-input"
        >
          <label>
            Hand Phone
          </label>
          <input
            aria-describedby="handphoneHelp"
            className="form-control"
            id="habdphoneInput"
            name="handphone"
            onChange={[Function]}
            placeholder="e.g 0773504973"
            type="text"
          />
        </div>
        <div
          className="password-input"
        >
          <label>
            Password
          </label>
          <input
            aria-describedby="passwordHelp"
            className="form-control"
            id="passwordlInput"
            name="password"
            onChange={[Function]}
            placeholder="Enter Password"
            type="password"
          />
        </div>
        <div
          className="confirm-input"
        >
          <label>
             Confirm Password
          </label>
          <input
            aria-describedby="passwordHelp"
            className="form-control"
            id="confirmPasswordlInput"
            name="confirmPassword"
            onChange={[Function]}
            placeholder="Confirm Password"
            type="password"
          />
        </div>
      </div>
      <div
        className="submit-button"
      >
        <button
          className="btn btn-primary submit-btn"
          id="signupButton"
          type="submit"
        >
          Sign Up 
          
        </button>
      </div>
    </form>
  </div>,
]
`;
